gradle/defaults.gradle

```diff
@@ -77,4 +68,13 @@ allprojects {
       }
     }
   }
+
+  // Repositories to fetch dependencies from.
+  repositories {
```

Comment:
Maybe check for a separate property "custom.snapshots.repo" and if it exists skip the "ApacheSnapshots" and define a "CustomRepository"? It's weird to just substitute the URL here. I'm not sure if a more elegant way of doing this exists (although there very likely must be one for folks behind corporate firewalls)...